### ISSUE TITLE
Deploying App-server with ssl-certificate-template is failing.

### ISSUE THREAD

**Issue by arulsiva21** [marklogic-community/ml-gradle#369]
I’m not sure if anyone reported this issue earlier, but I’m facing this when I try to update the default app-servers to https. 

Here is the step to reproduce this issue.

Step 1: Bring UP marklogic Server.
Step2:  Update Default app-servers config files to attach ssl-certificate-template.
Step 3: Order the app-servers config files appropriately in order to avoid manage app-server to be deployed first. Here is the order 01App-services, 02HealthCheck, 03Admin, 03Manage.
Step 4. Deploy using mlDeploy command.

Since 8001 port is being used for timestamp to see if server is healthy. Deployment is not getting done successfully as it’s trying to hit 8001 as http for health check, but 8001 (Admin) app-server was updated as https on the same Deploy. 

http://localhost:8001/admin/v1/timestamp

Please let us if know if there is a way to fix this issue by running mldeploy without breaking app-server deploy project.

**Comment by rjrudin** [comment]
I haven't heard of anyone making updates to the default app servers as part of an application deployment.

To me, these sound like steps that are part of a one-time initialization of an environment, which you would then deploy your application into via mlDeploy. 

You can make that initialization part of mlDeploy, but I would do it via a custom Gradle task that mlDeploy depends on. It sounds like that task would do the following:

# Install the certificate template
# Update each of the default servers to use the certificate template
# Then mlDeploy will run, and properties would be configured (e.g. mlAdminScheme, mlManageScheme) so that HTTPS is used for every connection

**Comment by rjrudin** [comment]
The default servers are created when MarkLogic is installed and then initialized. My recommendation is to use a separate process for then further initializing the environment, which would include creating the certificate template - e.g. via http://docs.marklogic.com/REST/POST/manage/v2/certificate-authorities . Then, use http://docs.marklogic.com/REST/PUT/manage/v2/servers/[id-or-name]/properties to update the template for each of the pre-installed app servers. 

You can use ml-app-deployer to simplify this via the CertificateTemplateManager and ServerManager classes within [a custom Gradle task](https://github.com/marklogic-community/ml-gradle/wiki/Writing-your-own-task). Or have your Ansible script do the work via curl calls. 

Basically, I see this work as separate from application deployment/undeployment (particularly since you don't want the pre-installed app servers undeployed). So I don't think it should be done via the mlDeploy task, but rather via a separate Gradle task or some other script that Ansible can invoke.
